jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -245,9 +251,22 @@ private static InsnNode simplifyArith(InsnNode insn) {
 			// fix 'c + (-1)' => 'c - (1)'
 			if (arith.getOp() == ArithOp.ADD && lit < 0) {
 				return new ArithNode(ArithOp.SUB,
-						arith.getResult(), insn.getArg(0),
+						arith.getResult(), arith.getArg(0),
 						InsnArg.lit(-lit, litArg.getType()));
 			}
+			InsnArg firstArg = arith.getArg(0);
+			if (arith.getOp() == ArithOp.XOR && firstArg.getType() == ArgType.BOOLEAN) {
+				if (lit == 1 && firstArg.isInsnWrap()) {
```

Comment:
So, you make it just for wrapped instructions, and it will not work for simple args (variables). There are test case:
```java
		public boolean test(boolean v) {
			return v ^ true; // expected 'return !v;'
		}
```
For this just do:
```java
if (lit == 1) {
  InsnNode not = new InsnNode(InsnType.NOT, 1);
  not.setResult(arith.getResult()); // note I return original result it is better because it will not broke SSA analysis
  not.addArg(firstArg);
  return not; 
}
```
For `lit == 0` you can return `MOVE` instruction